var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -227,6 +232,7 @@ def edit(self, spec, prefix):
             'nvhpc': ['-fast'],
             'cray': ['-O2'],
             'xl': ['-O3'],
+            'aocc': ['-O3', '-ffast-math', '-Hx,47,0x10000008', ],
```

Comment:
Please don't use `-ffast-math`. Been there. Done that.
And especially not if it means adding some undocumented internal flag to remedy whatever the optimization is doing.

Reply:
If the tests pass... in some cases you need ffast-math to get auto-vectorization.

Reply:
@dev-zero 
- “Hx…..” is a bug-fix required if -O2 or -O3 is used with AOCC 2.2 & 2.3. we are working this bug, you may not see this flag in upcoming AOCC releases.
- by dropping `-ffast-math` we are losing performance

Reply:
So, will the flag be still there and work in those future releases of AOCC or should this be guarded by a version check on the compiler from the start?
About performance, I guess it increases performance, but by how much, and on what?
Please note that the `tests` in CP2K are regression tests and will not catch numerical errors which can accumulate over long running simulations.

Reply:
Removed `-fast-math` and `-Hx` .
`-O3 -ffast-math -Hx` would be 6% better performant over `O1`. Tests with H2O-dft-ls show Energy results diverge at 9th decimal place.